index.bs

```diff
@@ -3288,27 +3404,67 @@ To <dfn>get the base network event data</dfn> given |request| and |redirect coun
      settings|' [=environment settings object/global object=], let |context id|
      be the [=browsing context id=] for that context.
 
+1. Let |redirect count| be |request|'s [=redirect count=].
+
 1. Let |timestamp| be a [=time value=] representing the current date and time in UTC.
 
+1. If |intercepts| is not [=list/empty=], let |is blocked| be true, otherwise
+   let |is blocked| be false.
+
 1. Let |params| be [=/map=] matching the <code>network.BaseParameters</code>
    production, with the <code>request</code> field set to |request data|, the
    |navigation| field set to <code>navigation</code>, the <code>context</code>
    field set to |context id|, the <code>timestamp</code> field set to
-   |timestamp|, and the <code>redirectCount</code> field set to |redirect count|.
+   |timestamp|, the <code>redirectCount</code> field set to |redirect count|, the
+   <code>isBlocked</code> field set to |is blocked|, and <code>intercepts</code>
+   field set to |intercepts| if |is blocked| is true, or omitted otherwise.
+
+1. Return |params|
+
+</div>
 
-1. Return (|related browsing contexts|, |params|)
+#### The network.Body Type #### {#type-network-Body}
+
+<pre class="cddl local-cddl remote-cddl">
+network.Body = network.StringBody / network.Base64Body;
+
+network.StringBody = {
+  type: "string",
+  value: text,
+}
+
+network.Base64Body = {
+  type: "base64",
+  value: text
+}
+</pre>
+
+The <code>network.Body</code> type represents a request or response body,
+possibly encoded.
+
+<div algorithm>
+To <dfn>get the body</dfn> given |protocol body|:
+
+1. If |protocol body| matches the <code>network.StringBody</code> production,
+   let |body| be [=UTF-8 encode=] |protocol body|["<code>value</code>"].
+
+1. Otherwise if |protocol body| matches the
+   <code>network.Base64Body</code>production. Let |body| be [=forgiving-base64
+   decode=] |protocol body|["<code>value</code>"].
+
+1. Return |body|
 
 </div>
 
 #### The network.Cookie Type #### {#type-network-Cookie}
 
 [=Remote end definition=] and [=local end definition=]
 
-<pre class="cddl local-cddl">
+<pre class="cddl local-cddl remote-cddl">
 network.Cookie = {
     name: text,
     ? value: text,
-    ? binaryValue: [ uint ]
+    ? binaryValue: [ *(0..255) ]
```

Comment:
Should `Cookie` make `value` and `binaryValue` mutually exclusive, like `Header` does?